src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -224,6 +224,39 @@ public ResponseEntity<ProgrammingExerciseStudentParticipation> resumeParticipati
         return ResponseEntity.ok().body(participation);
     }
 
+    /**
+     * PUT exercises/:exerciseId/request-feedback: Requests manual feedback for the latest participation
+     *
+     * @param exerciseId of the exercise for which to resume participation
+     * @param principal  current user principal
+     * @return ResponseEntity with status 200 (OK)
+     */
+    @PutMapping("exercises/{exerciseId}/request-feedback")
+    @PreAuthorize("hasRole('USER')")
+    @FeatureToggle(Feature.ProgrammingExercises)
+    public ResponseEntity<ProgrammingExerciseStudentParticipation> requestFeedback(@PathVariable Long exerciseId, Principal principal) {
+        log.debug("REST request for feeback request: {}", exerciseId);
+        var programmingExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exerciseId);
+        var participation = programmingExerciseParticipationService.findStudentParticipationByExerciseAndStudentId(programmingExercise, principal.getName());
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+
+        checkAccessPermissionOwner(participation, user);
+
+        var currentDate = now();
+
+        var participationIndividualDueDate = participation.getIndividualDueDate();
+        if (Objects.nonNull(participationIndividualDueDate) && currentDate.isAfter(participationIndividualDueDate)) {
+            throw new IllegalArgumentException("Request has already been sent");
```

Comment:
**CR**: That should be turned into a user-facing warning box that is able to use a translation key for the client via the `BadRequestException`.

Reply:
That behavior shouldn't be possible as the button to submit will be disabled.

Reply:
```suggestion
            throw new BadRequestException("Request has already been sent");
```
Okay, then it still should be a bad request exception to return the correct HTTP error code to indicate that the issue is with the data the client sent instead of an internal server error.
Same for the other `throw` above.